components/host-dashboard/UnhostAccountModal.js

```diff
@@ -0,0 +1,172 @@
+import StyledButton from '../StyledButton';
+import StyledLink from '../StyledLink';
+import StyledModal, { CollectiveModalHeader, ModalBody, ModalFooter } from '../StyledModal';
+import StyledTextarea from '../StyledTextarea';
+import { Label, P } from '../Text';
+import { TOAST_TYPE, useToasts } from '../ToastProvider';
+
+const unhostAccountMutation = gqlV2/* GraphQL */ `
+  mutation UnhostAccount($account: AccountReferenceInput!, $message: String) {
+    removeHost(account: $account, message: $message) {
+      id
+      slug
+      name
+      ... on AccountWithHost {
+        host {
+          id
+        }
+      }
+    }
+  }
+`;
+
+const UnhostAccountModal = ({ collective, host, ...props }) => {
+  const intl = useIntl();
+  const { addToast } = useToasts();
+  const [message, setMessage] = useState('');
+  const [unhostAccount, { loading }] = useMutation(unhostAccountMutation, {
+    context: API_V2_CONTEXT,
+    variables: {
+      account: {
+        id: collective.id,
+      },
+      message,
+    },
+    refetchQueries: ['HostDashboardHostedCollectives'],
+    awaitRefetchQueries: true,
+  });
+
+  return (
+    <StyledModal maxWidth={432} trapFocus {...props}>
+      <CollectiveModalHeader collective={collective} mb={3} />
+      <ModalBody>
+        <div>
+          <P fontSize="16px" fontWeight="700" lineHeight="24px" color="red.900" mb={2}>
+            <FormattedMessage
+              defaultMessage="Are you sure you want to un-host {collectiveName}?"
+              values={{ collectiveName: collective.name }}
+            />
+          </P>
+          <P fontSize="14px" lineHeight="20px" color="black.700" mb={2}>
+            <FormattedMessage
+              defaultMessage="{fiscalHostName} will no longer be the Fiscal Host for this Collective. Un-hosting this Collective means no longer holding funds for them, neither managing tax compliance and accounting."
+              values={{ fiscalHostName: host.name }}
+            />
+          </P>
+          {collective.stats.balance.valueInCents > 0 && (
+            <MessageBox type="warning" mb={2}>
+              <FormattedMessage
+                defaultMessage="The Collective's balance must be zero to un-host, including its Events or Projects. There is a remaining balance of {collectiveBalanceAmount}. You can pay out these funds by <Link>processing expenses.</Link>"
+                values={{
+                  collectiveBalanceAmount: formatCurrency(
+                    collective.stats.balance.valueInCents,
+                    collective.stats.balance.currency,
+                    { locale: intl.locale },
+                  ),
+                  Link: value => (
+                    <StyledLink
+                      color="black.800"
+                      href="https://docs.opencollective.com/help/collectives/collective-settings/zero-collective-balance"
+                      textDecoration="underline"
+                      openInNewTab
```

Comment:
:ice_cream: (personal preference): I'm not convinced we should open this one in a new tab, we do want people to switch context if they need to empty the balance.

```suggestion
                      as={Link}
```

Reply:
The link takes the user to the docs outside the main site, from the docs it might have a number of ways of achieving an empty balance so we are not guiding them through it here.

Reply:
You're right! I somehow misread the link and thought it was leading to the advanced settings